### PR TITLE
feat(BreakpointHooks): Port breakpoint hooks from Monolith into Gamut Labs

### PR DESCRIPTION
**Pull request by rogervera**
## Overview
These [hooks are currently in the monolith](https://github.com/codecademy-engineering/Codecademy/blob/master/webpack/assets/portal/libs/breakpoints.ts ), but now there’s a need for it in Static Sites - particularly for the time-sensitive BTS campaign. After discussion in [our internal Slack](https://codecademy.slack.com/archives/CQ98K8PT6/p1598458877049100), we are going with having this in **Gamut Labs** for now.

### PR Checklist

- [x] Related to JIRA ticket: [REACH-238](https://codecademy.atlassian.net/browse/REACH-238)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change (No tests, see discussion in original PR https://github.com/codecademy-engineering/Codecademy/pull/17702#discussion_r430765114)

### Description
These hooks help us dynamically structure the page based on the breakpoint. You can simply call `useBreakpoint()` to get the current breakpoint.

### CODE REVIEW

**Comment by rogervera** [comment]
Josh you had suggested no tests in the original PR since the logic became simple after we went with the `react-use` package [which already includes test](https://github.com/streamich/react-use/blob/master/tests/createBreakpoint.test.ts) for `createBreakpoint`), but **codecov** is upset about this! What do you suggest to do?

https://github.com/codecademy-engineering/Codecademy/pull/17702#discussion_r430765114

Screen Shot 2020-08-27 at 2 39 21 PM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@2.0.7-alpha.da2764.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f4801013b7f7900ccbe42fc--gamut-preview.netlify.app>https://5f4801013b7f7900ccbe42fc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f4801013b7f7900ccbe42fc>Deploy Logs</a></p>

File: packages/gamut-labs/src/lib/breakpointHooks/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+) as unknown) as BreakpointMinWidths;
+
+/**
+ * useBreakpoint will provide the current breakpoint to a React component.
+ *
+ * If you think you need this, make sure to think twice.  It is really nice
+ * to allow your styles layer to concern itself about the appearance of a
+ * component at different screen sizes, and not have to maintain two different
+ * components that could drift from eachother.  That being said, sometimes
+ * The content, layout, and/or functionality of a mobile page/component can
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/lib/breakpointHooks/index.ts:20]: heh, just noticed...

```suggestion
 * the content, layout, and/or functionality of a mobile page/component can
```

Comment (rogervera) [reply packages/gamut-labs/src/lib/breakpointHooks/index.ts:20]: 🧐

**Comment by rogervera** [comment]
> Codecov can shove it.
> 
> If you wanted to, you could add unit tests that these don't crash? But lmk if you want me to merge for you as admin.

@JoshuaKGoldberg I'd appreciate a good ol' Admin merge!

**Comment by JoshuaKGoldberg** [comment]
@rogervera do it your dang self!

_(just made you an owner, enjoy!)_

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@2.0.8-alpha.5d1f05.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f480a8ced8e98889a8c8f09--gamut-preview.netlify.app>https://5f480a8ced8e98889a8c8f09--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f480a8ced8e98889a8c8f09>Deploy Logs</a></p>

**Pull request merged**